addons/l10n_in_edi_ewaybill/models/account_edi_format.py

```diff
@@ -98,8 +98,12 @@ def _check_move_configuration(self, move):
                 error_message.append(_("- Transport document number and date is required when Transportation Mode is Rail,Air or Ship"))
         if error_message:
             error_message.insert(0, _("The following information are missing on the invoice (see eWayBill tab):"))
-        goods_lines = move.invoice_line_ids.filtered(lambda line: not (line.display_type in ('line_section', 'line_note', 'rounding') or line.product_id.type == "service"))
-        if not goods_lines:
+        invoice_lines = move.invoice_line_ids.filtered(lambda line: not (line.display_type in ('line_section', 'line_note', 'rounding')))
+        if any(not line.l10n_in_hsn_code for line in invoice_lines):
+            error_message.append(_("HSN/SAC code is missing on Invoice Line"))
```

Comment:
On which line?
If you have the same error multiple times on the same invoice, this will make a lot of noise

Reply:
Hello @william-andre ,
If more than one line does have not HSN then also warning will display only once because i checked with `any`. and then with a click of a button user can see all lines with an empty tax or HSN.

Thank you.